### PR TITLE
Enchanced docking behaviour for airlock_pump

### PR DESCRIPTION
**Pull request by Archemagus**
## About The Pull Request
Now after docking pump measure pressure on docked airlock, if it's in safe range both airlock sides will open for easy boarding process, instead just switching them like access airlocks

If docked entity pressure is lower then internal, then airlock will cycle only to external pressure, not to vacuum.
Also if opening airlocks, eash door also measures external pressure, to safe the day from bad pilot, when half of doors docked and other half still looking into space.

With new qol features pump now automatically detects docking\undocking to minimize vacuuming and extra headbumps.

Described behaviours tested, and work even for airlocks installed on shuttles, evac included, but night coding is messy, so a review will be nice.

## Why It's Good For The Game

If engis decide to build custom departure this feature should prevent crew from screaming "Malf AI" and speed up boarding\unboarding process (May be after this people will start to love those pumps)

## Changelog
:cl:
qol: Airlock_pump now dock with shuttles in more pleasant way
/:cl:

### CODE REVIEW

**Comment by Archemagus** [comment]
Example with docked evac shuttle, and not-open-airlocks-in-space feature

image

File: code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm [hunk 1]
```diff
@@ -125,7 +127,29 @@
 /obj/machinery/atmospherics/components/unary/airlock_pump/post_machine_initialize()
 	. = ..()
 	set_links()
-
+	// If we are on docked shuttle - setup docking variables
+	// Example - 'build your own shuttle' evac vessel
+	var/turf/local_turf = get_turf(src)
+	if (cycling_set_up && isshuttleturf(local_turf))
```
Comment (Ghommie) [inline code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm:133]: early return?

File: code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm [hunk 2]
```diff
@@ -323,6 +363,103 @@
...
+		if (airlocks_animating)	// Should (almost) never happened
+			say("ERROR: D11. Please re-initiate docking sequence.")
+			return
+
+	if (on)
+		// You can't go there, there is a shuttle now
+		if (pump_direction == ATMOS_DIRECTION_SIPHONING)
+			stop_cycle("Cycling sequence overriden by docking sequence.", TRUE)
+			start_cycle(ATMOS_DIRECTION_RELEASING)
+	else
```
Comment (Ghommie) [inline code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm:388]: Early return to reduce indentation.

File: code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm [hunk 3]
```diff
@@ -323,6 +363,103 @@
...
+		return FALSE
+	// No power handles by stop_cycle pretty good
+	else if (!powered())
+		return FALSE
+
+	var/turf/local_turf = get_turf(src)
+	var/tile_air_pressure = max(0, local_turf.return_air().return_pressure())
+	pressure_delta = internal_pressure_target - tile_air_pressure
+	// Chamber is pressurised too
+	if(pressure_delta <= allowed_pressure_error)
```
Comment (Ghommie) [inline code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm:418]: Ditto

**Comment by MTandi** [comment]
This pr has broken a few things. In particular, the setup when the shuttle is docked roundstart.

**Pull request merged**